lib/rucio/core/replica.py

```diff
@@ -3595,27 +4035,31 @@ def set_tombstone(rse_id, scope, name, tombstone=OBSOLETE, *, session: "Session"
     :param session: database session in use.
     """
     stmt = update(models.RSEFileAssociation).where(
-        and_(
-            models.RSEFileAssociation.rse_id == rse_id,
-            models.RSEFileAssociation.name == name,
-            models.RSEFileAssociation.scope == scope,
-            ~exists().where(
-                and_(
-                    models.ReplicaLock.rse_id == rse_id,
-                    models.ReplicaLock.name == name,
-                    models.ReplicaLock.scope == scope,
-                )
-            )
-        )
-    )\
-        .prefix_with("/*+ index(REPLICAS REPLICAS_PK) */", dialect='oracle')\
-        .execution_options(synchronize_session=False)\
-        .values(tombstone=tombstone)
-    result = session.execute(stmt)
+        and_(models.RSEFileAssociation.rse_id == rse_id,
+             models.RSEFileAssociation.name == name,
+             models.RSEFileAssociation.scope == scope,
+             ~exists().where(
+                 and_(models.ReplicaLock.rse_id == rse_id,
+                      models.ReplicaLock.name == name,
+                      models.ReplicaLock.scope == scope)))
+    ).prefix_with(
+        "/*+ INDEX(REPLICAS REPLICAS_PK) */", dialect='oracle'
+    ).values({
+        models.RSEFileAssociation.tombstone: tombstone
+    }).execution_options(
+        synchronize_session=False
+    )
 
-    if result.rowcount == 0:
+    if session.execute(stmt).rowcount == 0:
         try:
-            session.query(models.RSEFileAssociation).filter_by(scope=scope, name=name, rse_id=rse_id).one()
+            stmt = select(
+                models.RSEFileAssociation.tombstone
+            ).where(
+                and_(models.RSEFileAssociation.rse_id == rse_id,
+                     models.RSEFileAssociation.name == name,
+                     models.RSEFileAssociation.scope == scope)
+            )
+            _ = session.execute(stmt).scalar_one()
```

Comment:
You don't need the _ = here, you can just do the session.execute

Reply:
Removed, now just
```
session.execute(stmt).scalar_one()
```